### PR TITLE
Intly 6508 remove smtp providers

### PR DESCRIPTION
**Pull request by austinmartinh**
## Overview

Jira: https://issues.redhat.com/browse/INTLY-6508

## Verification

- Clone this branch
- Run `make cluster/prepare`
- Run `make cluster/seed/workshop/redis`
- Run `make cluster/seed/workshop/postgres`
- Run the operator
- Ensure resources create successfully

### CODE REVIEW

File: pkg/providers/config.go [hunk 1]
```diff
@@ -27,10 +27,9 @@ var DefaultConfigNamespace, _ = k8sutil.GetWatchNamespace()
 
 //DeploymentStrategyMapping Deployment Strategy Map
 type DeploymentStrategyMapping struct {
```
Comment (ciaranRoche) [inline pkg/providers/config.go]: in this directory can we update ` buildDefaultConfigMap()`

File: pkg/providers/aws/config.go [hunk 2]
```diff
@@ -91,22 +81,6 @@ func (m *ConfigMapConfigManager) ReadStorageStrategy(ctx context.Context, rt pro
...
 
-func (m *ConfigMapConfigManager) ReadSMTPCredentialSetStrategy(ctx context.Context, tier string) (*StrategyConfig, error) {
-	stratCfg, err := m.getTierStrategyForProvider(ctx, string(providers.SMTPCredentialResourceType), tier)
-	if err != nil {
-		return nil, errorUtil.Wrapf(err, "failed to get tier to strategy mapping for resource type %s", string(providers.BlobStorageResourceType))
-	}
-	return stratCfg, nil
-}
-
-func (m *ConfigMapConfigManager) GetDefaultRegionSMTPServerMapping() map[string]string {
```
[the remark this answers is not in the document]
Comment (austinmartinh) [reply pkg/providers/aws/config.go]: Ok

File: pkg/providers/config.go (hunk 1 above)
Comment (austinmartinh) [reply pkg/providers/config.go]: Yep, missed that

**Pull request merged**